packages/fetchai/skills/tac_control_contract/helpers.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the helpers methods for the controller agent."""
+
+import math
+import random
+from typing import Dict, List, Tuple, cast
+
+import numpy as np
+
+from aea.contracts.ethereum import Contract
+
+QUANTITY_SHIFT = 1  # Any non-negative integer is fine.
+TOKEN_TYPE = 2
+GAME_CURRENCY_TYPE = 2
+GAME_CURRENCY_ID = 2
```

Comment:
There will be an id clash in the current implementation. I realise we `do` need to feed in the good_ids to the `generate_token_ids` function. However, we should do it as an optional param.

Reply:
So something like 
``` python
def generate_token_id(token_type: str, token_nb: Optional[int], token_raw_ids: Optional[List[int]])
```

Reply:
Then if the `token_nb` is present, we do as implemented. If the `token_raw_ids` are present, we use these to feed into the helper function. That way, the same contract can be reused and new tokens can be minted.

Reply:
You can set `GAME_CURRENCY_ID = 1` by default and `ROW_TOKEN_IDS=[1,2,3,4,5,6,7,8,9,10]` by default. Both should be configurable so the same contract can be reused.

Reply:
Ok this is no longer relevant